RecoTracker/MkFit/plugins/createPhase1TrackerGeometry.cc

```diff
@@ -0,0 +1,372 @@
+//-------------------
+// Phase1 tracker geometry
+//-------------------
+
+#include "Config.h"
+#include "Debug.h"
+#include "TrackerInfo.h"
+#include "mkFit/IterationConfig.h"
+#include "mkFit/HitStructures.h"
+
+#include "HitSelectionWindowsPhase1.h"
+
+#include <functional>
+
+using namespace mkfit;
+
+namespace {
+#include "createPhase1TrackerGeometryAutoGen.acc"
+
+  void setupSteeringParamsIter0(IterationConfig &ic) {
+    ic.m_region_order[0] = TrackerInfo::Reg_Transition_Pos;
+    ic.m_region_order[1] = TrackerInfo::Reg_Transition_Neg;
+    ic.m_region_order[2] = TrackerInfo::Reg_Endcap_Pos;
+    ic.m_region_order[3] = TrackerInfo::Reg_Endcap_Neg;
+    ic.m_region_order[4] = TrackerInfo::Reg_Barrel;
+
+    {
+      SteeringParams &sp = ic.m_steering_params[TrackerInfo::Reg_Endcap_Neg];
+      sp.reserve_plan(3 + 3 + 6 + 18);
+      sp.fill_plan(0, 1, false, true);  // bk-fit only
+      sp.append_plan(2, true);          // pick-up only
+      sp.append_plan(45, false);
+      sp.append_plan(46, false);
+      sp.append_plan(47, false);
+      sp.fill_plan(48, 53);  // TID,  6 layers
+      sp.fill_plan(54, 71);  // TEC, 18 layers
+      sp.finalize_plan();
+    }
+    {
+      SteeringParams &sp = ic.m_steering_params[TrackerInfo::Reg_Transition_Neg];
+      sp.reserve_plan(3 + 4 + 6 + 6 + 8 + 18);
+      sp.fill_plan(0, 1, false, true);  // bk-fit only
+      sp.append_plan(2, true);
+      sp.append_plan(3, false);
+      sp.append_plan(45, false);
+      sp.append_plan(46, false);
+      sp.append_plan(47, false);
+      sp.fill_plan(4, 9);    // TIB,  6 layers
+      sp.fill_plan(48, 53);  // TID,  6 layers
+      sp.fill_plan(10, 17);  // TOB,  8 layers
+      sp.fill_plan(54, 71);  // TEC, 18 layers
+      sp.finalize_plan();
+    }
+    {
+      SteeringParams &sp = ic.m_steering_params[TrackerInfo::Reg_Barrel];
+      sp.reserve_plan(3 + 1 + 6 + 8);
```

Comment:
why 3+1 ? (phase0 + 1 ?)

Reply:
Yes, this is historical.

Reply:
since mkFit will never be used with phase0 geometry, do you mind to at least group into 4 ?

Reply:
Yes, the `3+1` can be replaced with `4`.